tests/components/blebox/test_cover.py

```diff
@@ -0,0 +1,381 @@
+)
+async def test_set_position(wrapper, hass, config):
+    """Test cover position setting."""
+
+    feature_mock, entity_id = wrapper
+
+    def initial_update():
+        feature_mock.state = 3  # closed
+
+    def set_position(position):
+        assert position == 99  # inverted
+        feature_mock.state = 1  # opening
+        # feature_mock.current = position
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    feature_mock.async_set_position = AsyncMock(side_effect=set_position)
+
+    await async_setup_entity(hass, config, entity_id)
+    assert hass.states.get(entity_id).state == STATE_CLOSED
+
+    feature_mock.async_update = AsyncMock()
+    await hass.services.async_call(
+        "cover",
+        SERVICE_SET_COVER_POSITION,
+        {"entity_id": entity_id, ATTR_POSITION: 1},
+        blocking=True,
+    )  # almost closed
+    assert hass.states.get(entity_id).state == STATE_OPENING
+
+
+async def test_unknown_position(shutterbox, hass, config):
+    """Test cover position setting."""
+
+    feature_mock, entity_id = shutterbox
+
+    def initial_update():
+        feature_mock.state = 4  # opening
+        feature_mock.current = -1
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OPEN
+    assert ATTR_CURRENT_POSITION not in state.attributes
+
+
+async def test_with_stop(gatebox, hass, config):
+    """Test stop capability is available."""
+
+    feature_mock, entity_id = gatebox
+    opening_to_stop_feature_mock(feature_mock)
+    feature_mock.has_stop = True
+
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    assert supported_features & SUPPORT_STOP
+
+
+async def test_with_no_stop(gatebox, hass, config):
+    """Test stop capability is not available."""
+
+    feature_mock, entity_id = gatebox
+    opening_to_stop_feature_mock(feature_mock)
+    feature_mock.has_stop = False
+
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    assert not supported_features & SUPPORT_STOP
+
+
+@pytest.mark.parametrize("wrapper", ALL_COVER_FIXTURES, indirect=["wrapper"])
+async def test_update_failure(wrapper, hass, config):
+    """Test that update failures are logged."""
+
+    feature_mock, entity_id = wrapper
+
+    feature_mock.async_update = AsyncMock(side_effect=blebox_uniapi.error.ClientError)
+    name = feature_mock.full_name
+
+    with patch("homeassistant.components.blebox._LOGGER.error") as error:
+        await async_setup_entity(hass, config, entity_id)
+
+        error.assert_has_calls([call("Updating '%s' failed: %s", name, ANY)])
+        assert isinstance(error.call_args[0][2], blebox_uniapi.error.ClientError)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)